pkg/controllers/backendrouting/backendrouting_controller.go

```diff
@@ -268,7 +268,7 @@ func (b *BackendRoutingReconciler) reconcileInClusterWithForwarding(ctx context.
 				return reconcile.Result{}, b.handleErr(ctx, br, backendsStatuses, routesStatuses, phase, v1alpha1.BackendUpgrading, err)
 			}
 
-			if backendsStatuses.Stable.Conditions.Ready == nil || *backendsStatuses.Stable.Conditions.Ready == false {
+			if !*backendsStatuses.Stable.Conditions.Ready {
```

Comment:
You should still check whether ready is nil
or you can use ptr.Deref to simplify the code